tests/parser.rs

```diff
@@ -1,91 +1,283 @@
-macro_rules! test_parser {
-    ($name:ident, $source:literal) => {
-        #[test]
-        fn $name() {
-            let ret = nolana::parser::Parser::new($source).parse();
-            insta::with_settings!({ omit_expression => true }, {
-                insta::assert_debug_snapshot!(ret);
-            });
-        }
-    };
+fn test_parser_helper(source: &str) -> String {
+    // This debug format is being used here to remove the need for
+    // insta::assert_debug_snapshot! and the need to fill in the filename,
+    // as the function name should give the filename for the snapshot
+    // this makes it follow the same filename layout as the rest of
+    // snapshots in snapshots directory
+    format!("{:?}", nolana::parser::Parser::new(source).parse())
```

Comment:
Good idea! Comment isn't needed since it's pretty common to not use debug assert. Most of the time it's `assert_snapshot` or one of the format-specific ones like JSON+serde (`assert_json_snapshot``).

Small thing: Directly import stuff from `nolana`. This also goes for `insta`.

```suggestion
use nolana::parser::Parser;
use insta::assert_snapshot;

fn test_parser_helper(source: &str) -> String {
    let result = Parser::new(source).parse();
    format!("{result:?}")
```